### ISSUE TITLE
[Doc]: Subheading links do not scroll to the appropriate page anchor

### ISSUE THREAD

**Issue by AnsonTran** [matplotlib/matplotlib#28082]
### Documentation Link

https://matplotlib.org/stable/users/index.html

### Problem

![image](https://github.com/matplotlib/matplotlib/assets/37683428/f9911a9c-99cf-44ec-b81e-9436887c58c8)

When clicking on the subheading links in the quick start guide, the page does not scroll to the appropriate part of the article. Additionally, if you go back to this page and click the subheading link again, it may lead to a different part of the page than before.

**Comment by tacaswell** [comment]
A very confusing thing to me that if you right-click "open in new tab" it _does_ work for me and in both cases the URL is the same!

https://matplotlib.org/stable/users/explain/quick_start.html#axis-scales-and-ticks
https://matplotlib.org/stable/users/explain/quick_start.html#axis-scales-and-ticks

**Comment by rcomer** [comment]
> A very confusing thing to me that if you right-click "open in new tab" it _does_ work for me and in both cases the URL is the same!

I see the opposite behaviour in iOS Safari!  It’s fine if I just tap the link but wrong if I “open in background“

@Impaler343 has also reported the problem and I reproduced on my work (Windows 10) laptop with both Chrome and Edge but not Firefox (though I only just installed Firefox to check).

**Comment by timhoffm** [comment]
Is there maybe some JavaScript doing funny things?

**Comment by jklymak** [comment]
```

<li class="toctree-l2"><a class="reference internal" href="explain/quick_start.html#parts-of-a-figure">Parts of a Figure</a></li>
```

In both Chrome and Safari these resolve to "https://matplotlib.org/stable/users/explain/quick_start.html#parts-of-a-figure" which is correct so far as I can tell

However when I click on this in Chrome, it does not go to the anchor, whereas it does in Safari.

In the source of the `quick_start.html`

```html
<section id="parts-of-a-figure">
<span id="figure-parts"></span><h2>Parts of a Figure<a class="headerlink" href="#parts-of-a-figure" title="Link to this heading">#</a></h2>
 <p>Here are the components of a Matplotlib Figure.</p>
...
</section>
```

Is it possible that Chrome doesn't recognize `<section>` as a place to find the id?  Google claims that should be OK:
https://developers.google.com/style/headings-targets

**Comment by jklymak** [comment]
I actually think this is a chrome bug.  If I do

```html
<section id="parts-of-a-figure">
<h1>Boooooo</h1>
Lots of text Lots of text Lots of text Lots of text Lots of text Lots of text Lots of text Lots of text
...
</section>
<section id="boo2">
    <h1>Boooooo 2</h1>
    Lots of text Lots of text Lots of text Lots of text Lots of text Lots of text Lots of text Lots of text
    .....
</section>
```
and navigate to `tmp/test.html#boo2` it goes to the anchor in Safari, and does not in Chrome.

**Comment by timhoffm** [comment]
If it's version dependent, it cannot be *only*  a chrome bug. The html looks identical AFAICS,

**Comment by jklymak** [comment]
Yes, fair enough - again still works fine in Safari, but not in Chrome

3.8 seemed to have added more indexing on the left column and better tracking of the document in the right-hand column.  Maybe Chrome is keying on those indices rather than in the body?
